packages/core/src/features/st-import.ts

```diff
@@ -0,0 +1,219 @@
+import { createFeature, FeatureContext, FeatureTransformContext } from './feature';
+import { generalDiagnostics } from './diagnostics';
+import type { Imported, ImportSymbol } from './types';
+import * as STSymbol from './st-symbol';
+import { plugableRecord } from '../helpers/plugable-record';
+import { ignoreDeprecationWarn } from '../helpers/deprecation';
+import { parseStImport, parsePseudoImport, parseImportMessages } from '../helpers/import';
+import { isCSSVarProp } from '../stylable-utils';
+import type { StylableMeta } from '../stylable-meta';
+import { rootValueMapping, valueMapping } from '../stylable-value-parsers';
+import path from 'path';
+import type { ImmutablePseudoClass, PseudoClass } from '@tokey/css-selector-parser';
+import type * as postcss from 'postcss';
+
+const dataKey = plugableRecord.key<Imported[]>('imports');
+
+export const diagnostics = {
+    ...parseImportMessages,
+    NO_ST_IMPORT_IN_NESTED_SCOPE() {
+        return `cannot use "@st-import" inside of nested scope`;
+    },
+    NO_PSEUDO_IMPORT_IN_NESTED_SCOPE() {
+        return `cannot use ":import" inside of nested scope`;
+    },
+    INVALID_CUSTOM_PROPERTY_AS_VALUE(name: string, as: string) {
+        return `invalid alias for custom property "${name}" as "${as}"; custom properties must be prefixed with "--" (double-dash)`;
+    },
+    FORBIDDEN_DEF_IN_COMPLEX_SELECTOR: generalDiagnostics.FORBIDDEN_DEF_IN_COMPLEX_SELECTOR,
```

Comment:
The diagnostics in `generalDiagnostics` are common to more than one feature. And I prefer tests will be agnostic to multiple diagnostics location and just use the feature's diagnostics.

The reason some places still use the general diagnostics directly is because these places still have code that wasn't extracted to a feature yet (mostly in the processor).